src/context/subscriber.js

```diff
@@ -0,0 +1,226 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {ContextNode} from './node';
+import {arrayOrSingleItemToArray} from '../types';
+import {throttleTail, tryCallback} from './scheduler';
+
+const EMPTY_ARRAY = [];
+const EMPTY_FUNC = () => {};
+
+/**
+ * Subscribes to the specified dependencies. The key API points:
+ * - The subscriber is only called when all its dependencies are satisfied.
+ * - If all its dependencies are satisfied at the time the subscriber is added,
+ *   it's called right away.
+ * - The subscriber is called whenever any of its dependencies change.
+ * - A subscriber can optionally return a cleanup function.
+ *
+ * @param {!Node} node
+ * @param {!ContextProp|!Array<!ContextProp>} deps
+ * @param {function(...?)} callback
+ */
+export function subscribe(node, deps, callback) {
+  deps = arrayOrSingleItemToArray(deps);
+  const id = callback;
+  const contextNode = ContextNode.get(node);
+  contextNode.subscribe(id, Subscriber, callback, deps);
+}
+
+/**
+ * Removes the subscriber prevoiously registered with `subscribe` API.
+ *
+ * @param {!Node} node
+ * @param {function(...?)} callback
+ */
+export function unsubscribe(node, callback) {
+  const id = callback;
+  const contextNode = ContextNode.get(node);
+  contextNode.unsubscribe(id);
+}
+
+/**
+ * The subscriber holder. It encapsulates the subscriber's dependencies,
+ * internal state, and cleanup functions.
+ *
+ * @package
+ */
+export class Subscriber {
```

Comment:
This is simply renamed and moved nearly as is from `Component` in the old `component.js`. No real changes, just removal. Diff is not smart enough here to connect the dots.

Reply:
See diff: https://gist.github.com/dvoytenko/172307cbf8e2bb2db559c8fb2ad218d6